### DISCUSSION TITLE
Modify plist in post-install

### DISCUSSION THREAD

**Discussion by tillkruss** [Homebrew/discussions#1637] [Tap maintenance and Homebrew development] [answered]
We are distributing a private Homebrew tap/formula and need to modify the plist of the [PHP formula](https://formulae.brew.sh/formula/php).

I'm able to read, but not write to `Formula["php"].plist_path` in the `post_install` method.

I've tried using `IO.popen`, `system`, etc, all result in:

```
Operation not permitted
```

Is this a limitation of the post-install method, or a restriction on not being able to modify the cellar of other formulas? Is there any way around this (for a private formula)?

For context, [we need to set](https://www.wefearchange.org/2018/11/forkmacos.rst.html) `OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` for the PHP process:

```bash
plutil -insert EnvironmentVariables -xml '<dict/>' ${plist}
plutil -replace EnvironmentVariables.OBJC_DISABLE_INITIALIZE_FORK_SAFETY -string YES ${plist}
```

**Comment by SMillerDev** [comment, marked as answer]
Homebrew formulae can't write to each other's directories. This is by design.
